lib/puppet/module.rb

```diff
@@ -428,22 +428,25 @@ def initialize_i18n
 
     locales_path = File.absolute_path('locales', path)
 
-    begin
-      Puppet::GettextConfig.initialize(locales_path, :po)
+    if Puppet::GettextConfig.initialize(locales_path, :po)
       Puppet.debug "#{module_name} initialized for i18n: #{GettextSetup.translation_repositories[module_name]}"
-    rescue
-      config_path = File.absolute_path('config.yaml', locales_path)
-      Puppet.debug "Could not find locales configuration file for #{module_name} at #{config_path}. Skipping i18n initialization."
+    else
+      if Puppet::GettextConfig.gettext_found?
+        config_path = File.absolute_path('config.yaml', locales_path)
```

Comment:
This part is a little confusing... 

The code for `Puppet::GettextConfig.initialize(locales_path, :po)` doesn't appear to be looking for `config.yaml`... but the underlying call to the setup gem (at https://github.com/puppetlabs/gettext-setup-gem/blob/643d6f32d152774a3ea540cd649c6d025d5d3ed2/lib/gettext-setup/gettext_setup.rb#L19) does that bit.

It looks like there are a number of other things that are necessary to init properly, so I'd suggest we make the debug message more generic.  As it stands it appears to be potentially misleading as `config.yaml` might exist, but other things may not...

It also looks like that message will never get hit anyhow, given the setup gem raises with a missing `config.yaml` per https://github.com/puppetlabs/gettext-setup-gem/blob/643d6f32d152774a3ea540cd649c6d025d5d3ed2/lib/gettext-setup/gettext_setup.rb#L27

Reply:
So this message actually does show up whenever a module does not have translations. But we never hit the exception in the gem because of this check: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/gettext/config.rb#L46. Instead the initialize just returns false if the locales directory isn't present. I agree the debug message could be more generic, maybe saying it can't find the whole folder, but I do like distinguishing between "can't find the gem" and "can't find the files".